cocos/platform/CCFileUtils.cpp

```diff
@@ -860,9 +860,11 @@ std::string FileUtils::fullPathFromRelativeFile(const std::string &filename, con
 void FileUtils::setSearchResolutionsOrder(const std::vector<std::string>& searchResolutionsOrder)
 {
     bool existDefault = false;
+    std::vector<std::string> searchResolutionOrderCopied = searchResolutionsOrder;
+
     _fullPathCache.clear();
     _searchResolutionsOrderArray.clear();
-    for(const auto& iter : searchResolutionsOrder)
+    for(const auto& iter : searchResolutionOrderCopied)
```

Comment:
Suppose this invocation: 
```c++
fileUtils->setSearchResolutionsOrder(_searchResolutionsOrderArray);
```
_searchResolutionsOrderArray will be cleared before the for loop.
The same situation could happen in setSearchPaths.